src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingAssessmentResource.java

```diff
@@ -0,0 +1,95 @@
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    private final ProgrammingExerciseService programmingExerciseService;
+
+    private final ProgrammingAssessmentService programmingAssessmentService;
+
+    private final ProgrammingSubmissionService programmingSubmissionService;
+
+    public ProgrammingAssessmentResource(AuthorizationCheckService authCheckService, UserService userService, ProgrammingExerciseService programmingExerciseService,
+            ProgrammingAssessmentService programmingAssessmentService, ProgrammingSubmissionService programmingSubmissionService) {
+        super(authCheckService, userService);
+        this.programmingExerciseService = programmingExerciseService;
+        this.programmingAssessmentService = programmingAssessmentService;
+        this.programmingSubmissionService = programmingSubmissionService;
+    }
+
+    /**
+     * Update an assessment after a complaint was accepted.
+     *
+     * @param submissionId the id of the submission for which the assessment should be updated
+     * @param assessmentUpdate the programing assessment update
+     * @return the updated result
+     */
+    @ResponseStatus(HttpStatus.OK)
+    @PutMapping("/programming-submissions/{submissionId}/assessment-after-complaint")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Result> updateProgrammingManualResultAfterComplaint(@RequestBody ProgrammingAssessmentUpdate assessmentUpdate, @PathVariable Long submissionId) {
+        log.debug("REST request to update the assessment of manual result for submission {} after complaint.", submissionId);
+        User user = userService.getUserWithGroupsAndAuthorities();
+        ProgrammingSubmission programmingSubmission = programmingSubmissionService.findOneWithEagerResultAndFeedback(submissionId);
+        Participation participation = programmingSubmission.getParticipation();
+        long exerciseId = participation.getExercise().getId();
+        ProgrammingExercise programmingExercise = programmingExerciseService.findOne(exerciseId);
+        checkAuthorization(programmingExercise, user);
+        if (!areManualResultsAllowed(programmingExercise)) {
+            return forbidden();
+        }
+
+        Result result = programmingAssessmentService.updateAssessmentAfterComplaint(programmingSubmission.getResult(), programmingExercise, assessmentUpdate);
+
+        // remove circular dependencies if the results of the participation are there
+        if (result.getParticipation() != null && Hibernate.isInitialized(result.getParticipation().getResults()) && result.getParticipation().getResults() != null) {
+            result.getParticipation().setResults(null);
+        }
+
+        if (result.getParticipation() != null && result.getParticipation() instanceof StudentParticipation
+                && !authCheckService.isAtLeastInstructorForExercise(programmingExercise, user)) {
```

Comment:
You might be able to replace this Block by using
```java
programmingSubmissionService.hideDetails(programmingSubmission);
```

Can you check if that is a possibility here, as I am not quite sure.

Reply:
No. This is used for submission and we never use it in this case.